src/ln/channelmanager.rs

```diff
@@ -1370,7 +1370,7 @@ impl ChainListener for ChannelManager {
 						Ok(res) => res,
 						Err(e) => {
 							log_error!(self, "Got error handling message: {}!", e.err);
-							//TODO: push e on events and blow up the channel (it has bad keys)
+							//TODO: handle WaitNewState or IgnoreMessage
```

Comment:
We don't need KeyError anymore here ! Just to fulfill the downstream HandleError with some WaitNewState (waiting for ChannelState::ChannelFunded) or other

Reply:
I dont think WaitNewState makes sense - maybe there's some confusion here but if you see a "bad key" from something the remote generated they are clearly misbehaving - if you could randomly hit bad keys on accident it'd be a sign that the underlying ECC crypto were broken, so we should treat it as any other clearly-misbehaving node.

Reply:
Not targeting the "bad key" err here but more the "Cannot get a ChannelAnnouncement until the channel funding has been locked" from get_channel_announcement which can be return upstream in block_connected. How should we handle this kind of case ?

Reply:
Ahh, I missed context, in any case, we can simplify this further, see #133.